sdk/core/azure-core/src/main/java/com/azure/core/models/ResponseError.java

```diff
@@ -117,4 +123,96 @@ ResponseError setErrorDetails(List<ResponseError> errorDetails) {
+            //
+            // - A ResponseError object
+            // - A ResponseError object wrapped in an "error" node.
+            JsonReader bufferedReader = reader.bufferObject();
+            bufferedReader.nextToken(); // Get to the START_OBJECT token.
+            while (bufferedReader.nextToken() != JsonToken.END_OBJECT) {
+                String fieldName = bufferedReader.getFieldName();
+                bufferedReader.nextToken();
+
+                if ("error".equals(fieldName)) {
+                    // If the ResponseError was wrapped in the "error" node begin reading it now.
+                    return readResponseError(bufferedReader);
+                } else {
+                    bufferedReader.skipChildren();
+                }
+            }
+
+            // Otherwise reset the JsonReader and read the whole JSON object.
+            return readResponseError(bufferedReader.reset());
+        });
+    }
+
+    private static ResponseError readResponseError(JsonReader jsonReader) throws IOException {
+        return jsonReader.readObject(reader -> {
+            String code = null;
+            boolean codeFound = false;
+            String message = null;
+            boolean messageFound = false;
+            String target = null;
+            ResponseInnerError innerError = null;
+            List<ResponseError> errorDetails = null;
+
+            while (reader.nextToken() != JsonToken.END_OBJECT) {
+                String fieldName = reader.getFieldName();
+                reader.nextToken();
+
+                if ("code".equals(fieldName)) {
+                    code = reader.getString();
+                    codeFound = true;
+                } else if ("message".equals(fieldName)) {
+                    message = reader.getString();
+                    messageFound = true;
+                } else if ("target".equals(fieldName)) {
+                    target = reader.getString();
+                } else if ("innererror".equals(fieldName)) {
+                    innerError = ResponseInnerError.fromJson(reader);
+                } else if ("details".equals(fieldName)) {
+                    errorDetails = reader.readArray(ResponseError::fromJson);
+                } else {
+                    reader.skipChildren();
+                }
+            }
+
+            if (codeFound && messageFound) {
+                return new ResponseError(code, message)
+                    .setTarget(target)
+                    .setInnerError(innerError)
+                    .setErrorDetails(errorDetails);
+            }
+
+            if (!codeFound && !messageFound) {
+                throw new IllegalStateException("Missing required properties: code, message");
+            } else if (!codeFound) {
+                throw new IllegalStateException("Missing required property: code");
+            } else {
+                throw new IllegalStateException("Missing required property: message");
+            }
```

Comment:
We can skip this as we don't enforce this for responses now.

Reply:
Will update this as well